src/server/launchpad/resources.js

```diff
@@ -9,11 +9,21 @@ import { normalizeURI } from './uri';
 /** The base class for objects retrieved from Launchpad's web service. */
 export class Resource {
   /** Initialize a resource with its representation and URI. */
-  init(client, representation, uri) {
-    this.lp_client = client;
-    this.uri = uri;
-    for (const key of Object.keys(representation)) {
-      this[key] = representation[key];
+  init(client, uri, representation) {
+    // defining internal properties as non-enumerable not to leak any secrets
+    Object.defineProperty(this, 'lp_client', {
```

Comment:
I made sure in other parts that `lp_client` is always inherited (via `extends Resource`), so shouldn't be enumerable, but as it contains very sensitive data it's better to make it explicitly non-enumerable in case we instantiate `Resource` directly at some point.